code/clue.js

```diff
@@ -38,22 +196,90 @@ const randomSelector = array => {
 }
 
 // CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+
+let mystery = {
+  killer: null,
+  weapons: null,
+  room: null
+};
+
 
 // This function will be invoked when you click on the killer card.
+
 const pickKiller = () => {
+
   // This will randomly select a killer from the suspects. And add that to the mystery object.
   mystery.killer = randomSelector(suspects)
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  const killerName = document.getElementById('killerName')
+  const killerOccupation = document.getElementById('killerOccupation')
+  const killerCard = document.getElementById('killerCard')
+  const killerImage = document.getElementById('killerImage');
+
+  killerCard.style.background = mystery.killer.color
+  killerName.innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+  killerOccupation.innerHTML = `${mystery.killer.occupation}`;
+  killerImage.src = mystery.killer.image;
+
+  // shuffel favorite weapon function.
+  const shuffleFavouriteWeapon = () => {
+  mystery.favouriteWeapon = randomSelector(weapons);
+  document.getElementById('favouriteWeapon').innerHTML = `Favourite weapon: ${mystery.favouriteWeapon.name}`;
+  }
+  shuffleFavouriteWeapon();
```

Comment:
Nice way to utilize the pickKiller function by adding the randomizeweapon function within!

Reply:
If you are going to define a function within another function, pay attention to indentation.  Also, the reasoning behind the decision.  

What's preventing you from declaring this function outside of `pickKiller`?  How could you give access to the `weapons` data without putting the function declaration in the same scope?